### PR TITLE
Set pre sorting to put null last. Fixes #247

### LINKED ISSUE #247: Fix pre version sort order
Currently in the version sort order something comes after absent for the `pre` field. According to point 11 in https://semver.org/ absent should come after something.

### PR DESCRIPTION
**Pull request by johnoliver**
# Checklist

- [ ] You added tests to cover the change
- [ ] `mvn clean install` build and test completes
- [ ] You changed or added to the documentation

### CODE REVIEW

File: adoptopenjdk-api-v3-models/src/test/kotlin/net/adoptopenjdk/api/VersionParserTest.kt [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 import net.adoptopenjdk.api.v3.parser.VersionParser
 import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNull
 import org.junit.jupiter.api.Assertions.assertTrue
 import org.junit.jupiter.api.DynamicTest
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.TestFactory
+import java.net.URLDecoder
+import java.nio.charset.Charset
+import java.util.*
```
Comment (karianna) [inline adoptopenjdk-api-v3-models/src/test/kotlin/net/adoptopenjdk/api/VersionParserTest.kt:16]: prefer explicit exports

**Review by karianna — APPROVED** [review]
LGTM - one nitpick

**Comment by karianna** [comment]
@johnoliver `/home/runner/work/openjdk-api-v3/openjdk-api-v3/adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/routes/packages/PackageEndpoint.kt:8:1: Unused import`

File: adoptopenjdk-api-v3-models/src/test/kotlin/net/adoptopenjdk/api/VersionParserTest.kt (hunk 1 above)
Comment (nickebbitt) [reply adoptopenjdk-api-v3-models/src/test/kotlin/net/adoptopenjdk/api/VersionParserTest.kt:16]: we should really have some IDE or `editorconfig` for this rule, I've seen you mention it at least 10 times :-)

it looks like it's an option here: https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties

but we appear to disable it for some reason https://github.com/AdoptOpenJDK/openjdk-api-v3/blob/master/.editorconfig#L944

Comment (johnoliver) [reply adoptopenjdk-api-v3-models/src/test/kotlin/net/adoptopenjdk/api/VersionParserTest.kt:16]: we can fix that, issue is default intellij config is to wildcard import java.* and some other stuff, and I am not sure if that can be changed from editorconfig

Comment (johnoliver) [reply adoptopenjdk-api-v3-models/src/test/kotlin/net/adoptopenjdk/api/VersionParserTest.kt:16]: possibly `ij_java_imports_layout`

**Review by nickebbitt — APPROVED** [review]

**Comment by karianna** [comment]
`AssetsResourceFeatureReleasePathSortOrderTest.doesSortObaySortMethod:34 expected:<[bar]> but was:<[foo]>`

**Comment by karianna** [comment]
@johnoliver I'm good to merge this if you're convinced this will help...

**Comment by karianna** [comment]
Closing in favour of newer PR

**Pull request closed without merging**